lib/Intlc/Backend/TypeScript/Compiler.hs

```diff
@@ -25,8 +25,14 @@ compileNamedExport x l k v =
 compileTypeof :: InterpStrat -> ICU.Message -> Text
 compileTypeof x = let o = fromStrat x in flip runReader o . typeof . fromMsg o
 
-reactImport :: Text
-reactImport = "import React, { ReactElement } from 'react'"
+isTypeScriptReact :: Translation -> Bool
+isTypeScriptReact (Translation _m  be) = case be of
+  TypeScript      -> False
+  TypeScriptReact -> True
+
+
+buildReactImport :: Dataset Translation -> Maybe Text
+buildReactImport = fmap (const "import React, { ReactElement } from 'react'") . find isTypeScriptReact
```

Comment:
We can move the string into a `where` clause to make the top-level algorithm more focused.

We're discarding the contents of the `Just` coming from `find`, so we could instead use `any`. In turn we'll need to change the code slightly to operate over `Bool`. There are a few different ways, here's one example:

```hs
flip pureIf txt . any isTypeScriptReact

-- or not pointfree, arguably more readable?
xs = pureIf (any isTypeScriptReact xs) txt
```

(where `txt` is in a `where` clause)